diff --git a/src/MicrosoftAuthManager.php b/src/MicrosoftAuthManager.php
index 4a1d139..9ae1b00 100644
--- a/src/MicrosoftAuthManager.php
+++ b/src/MicrosoftAuthManager.php
@@ -70,7 +70,7 @@ class MicrosoftAuthManager extends OAuth2Manager {
 
     // Returns the URL where user will be redirected.
     return $this->client->getAuthorizationUrl([
-      'scope' => $scopes,
+      'scope' => ['https://graph.microsoft.com/User.Read'],
     ]);
   }
 
diff --git a/src/Plugin/Network/MicrosoftAuth.php b/src/Plugin/Network/MicrosoftAuth.php
index 5544bbf..6b495dc 100644
--- a/src/Plugin/Network/MicrosoftAuth.php
+++ b/src/Plugin/Network/MicrosoftAuth.php
@@ -52,6 +52,9 @@ class MicrosoftAuth extends NetworkBase implements MicrosoftAuthInterface {
         'clientId' => $settings->getAppId(),
         'clientSecret' => $settings->getAppSecret(),
         'redirectUri' => Url::fromRoute('social_auth_microsoft.callback')->setAbsolute()->toString(),
+        'urlAuthorize'              => 'https://login.microsoftonline.com/common/oauth2/v2.0/authorize',
+        'urlAccessToken'            => 'https://login.microsoftonline.com/common/oauth2/v2.0/token',
+        'urlResourceOwnerDetails'   => 'https://graph.microsoft.com/oidc/userinfo'
       ];
 
       // Proxy configuration data for outward proxy.
